util/jsonschema.py

```diff
@@ -1,42 +1,61 @@
-from typing import Any, Dict, List
-
-JSONSchema = Any
-Str = {"type": "string"}
-Int = {"type": "integer"}
-Bool = {"type": "boolean"}
-
-
-def nullable(schema: JSONSchema) -> JSONSchema:
-    return {"anyOf": [schema, {"type": "null"}]}
-
-
-def Obj(**properties: Dict[str, JSONSchema]) -> JSONSchema:
-    return {
-        "type": "object",
-        "properties": properties,
-        "additionalProperties": False,
-        "required": list(properties.keys()),
-    }
-
-
-def Enum(values: List[str]) -> JSONSchema:
-    return {
-        "type": "string",
-        "enum": values,
-    }
-
-
-def IntRange(
-    minimum: int, maximum: int, exclusiveMaximum=False, exclusiveMinimum=False
-):
-    if exclusiveMinimum:
-        minimum += 1
-    if exclusiveMaximum:
-        maximum -= 1
-
-    assert minimum <= maximum
-    return {
-        "type": "integer",
-        "minimum": minimum,
-        "maximum": maximum,
-    }
+import jsonschema
+from typing import Any, Dict, List, Union
+
+JSONSchema = Dict[str, Any]
+
+
+def validate(instance: Any, schema: JSONSchema):
+    validate_schema(schema)
+    jsonschema.validate(instance=instance, schema=schema)
```

Comment:
Do you think it's worth calling the `jsonschema` library's validator on the schema before calling our validator? I was thinking it might be able to catch some more basic structural errors and provide good error messages. It looks like this method would do it: https://python-jsonschema.readthedocs.io/en/stable/validate/#jsonschema.IValidator.check_schema

Reply:
I think I got it working, though it's clearly not an API they really intend for outside folks to use for this purpose.

Reply:
😳